lib/util/makeRequest.ts

```diff
@@ -82,45 +63,38 @@ const promisifiedRequest = (options: Options) =>
  * @return {Promise} promise - A promise that resolves to the responseBody or rejects to an error.
  */
 export async function makeRequest(
-  options: Options,
+  url: string,
   reportError: jsonParser.ReportError
 ): Promise<SwaggerObject> {
-  return retry(
-    async (bail, retryNr) => {
-      const { err, response, responseBody } = await promisifiedRequest(options)
-
-      if (err) {
-        const message = `Request to ${
-          options.url
-        } failed with error ${err} on retry number ${retryNr}.`
-
-        throw new Error(message)
-      }
+  return retry<SwaggerObject>(
+    async (bail, retryNr): Promise<SwaggerObject> => {
+      try {
+        const response = await fetch(url)
+        const body = await response.text()
 
-      if (options.errorOnNon200Response && response.statusCode !== 200) {
-        const msg = `StatusCode: "${
-          response.statusCode
-        }", ResponseBody: "${responseBody}."`
+        if (response.status !== 200) {
+          const msg = `StatusCode: "${
+            response.status
+          }", ResponseBody: "${body}."`
 
-        return bail(new Error(msg))
-      }
+          return bail(new Error(msg)) as any
```

Comment:
and why are we bailing on a bad code ?

Reply:
maybe its some temp issue where its failing, since its not really an api call I guess probably its fine, but I would just retry in this case also

Reply:
hm, retry even on 404?

Reply:
well, i would say no, but the api returns 404 for some weird reasons so your call. and if for some reason github returns 404 because the resource is not accessible for a little while retryig wont harm us